File: 1732. Find the Highest Altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        // Current altitude of the biker
        int currentAlti = 0;
        
        // Highest altitude reached so far
        int maximumAlti = 0;
        
        // Traverse all altitude changes
        for (int change : gain) {
            // Update current altitude
            currentAlti += change;
            
            // Update maximum altitude if needed
            maximumAlti = Math.max(maximumAlti, currentAlti);
        }
        
        return maximumAlti;
    }
}

File: 1732. Find the Highest Altitude.py
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        # Current altitude of the biker
        current_alti = 0
        
        # Highest altitude reached so far
        maximum_alti = 0
        
        # Traverse all altitude changes
        for change in gain:
            # Update current altitude
            current_alti += change
            
            # Update maximum altitude if needed
            maximum_alti = max(maximum_alti, current_alti)
        
        return maximum_alti
